### PR TITLE
feature(new integration): Zapier Onboarding

### PR DESCRIPTION
**Pull request by rohithkaza**
## Description of the change

> Onboarding Zapier Cloud Mode Destination

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: cdk/zapier/transform.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  rudderContext.zapUrl = destConfig.zapUrl;
+
+  // if mapping(event to ZapUrl) is present in the dashboard
+  Object.keys(eventsMap).forEach(key => {
+    if (
+      (message?.type === "track" && key === message?.event) ||
+      ((message?.type === "page" || message?.type === "screen") &&
+        key === message?.name)
+    ) {
+      rudderContext.zapUrl = eventsMap[key];
```
Comment (ItsSudip) [inline cdk/zapier/transform.js:21]: Can you check with segment if the event name or page name getting matched with any URL then are they sending the event only to the matched URL or both the matched and default URL?

File: cdk/zapier/transform.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  // default Zap URL
+  rudderContext.zapUrl = destConfig.zapUrl;
+
+  // if mapping(event to ZapUrl) is present in the dashboard
+  Object.keys(eventsMap).forEach(key => {
+    if (
+      (message?.type === "track" && key === message?.event) ||
+      ((message?.type === "page" || message?.type === "screen") &&
+        key === message?.name)
+    ) {
```
Comment (ItsSudip) [inline cdk/zapier/transform.js:20]: Let's don't merge the track and page/screen URLs.

**Review by ItsSudip — CHANGES_REQUESTED** [review]

File: cdk/zapier/transform.js (hunk 1 above)
Comment (rohithkaza) [reply cdk/zapier/transform.js:21]: Yes, they are only sending the event to the matched URL.

File: cdk/zapier/transform.js (hunk 2 above)
Comment (rohithkaza) [reply cdk/zapier/transform.js]: Done

File: cdk/zapier/transform.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+  // page/screen event
+  if (message?.type === "page" || message?.type === "screen") {
+    // checking if the event is present page/screen events mapping
+    Object.keys(pageScreenEventsMap).forEach(key => {
+      if (key === message?.name) {
+        // if present, we are updating the zapUrl(with the one specified in the mapping)
+        rudderContext.zapUrl = pageScreenEventsMap[key];
+      }
+    });
```
Comment (ItsSudip) [inline cdk/zapier/transform.js:34]:
```suggestion
    const pageScreenName = message?.name; 
    if(pageScreenEventsMap[pageScreenName]){
        rudderContext.zapUrl = pageScreenEventsMap[pageScreenName];
    }
```

File: cdk/zapier/transform.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+  // track event
+  if (message?.type === "track") {
+    // checking if the event is present track events mapping
+    Object.keys(trackEventsMap).forEach(key => {
+      if (key === message?.event) {
+        // if present, we are updating the zapUrl(with one specified in the mapping)
+        rudderContext.zapUrl = trackEventsMap[key];
+      }
+    });
```
Comment (ItsSudip) [inline cdk/zapier/transform.js:23]:
```suggestion
    const eventName = message?.event; 
    if(trackEventsMap[eventName]){
        rudderContext.zapUrl = trackEventsMap[eventName];  
    }
```

**Review by ItsSudip — CHANGES_REQUESTED** [review]

File: cdk/zapier/transform.js (hunk 3 above)
Comment (rohithkaza) [reply cdk/zapier/transform.js]: Done

File: cdk/zapier/transform.js (hunk 4 above)
Comment (rohithkaza) [reply cdk/zapier/transform.js]: Done

**Review by ItsSudip — APPROVED** [review]

**Review by ujjwal-ab — DISMISSED** [review]

**Review by ItsSudip — APPROVED** [review]

**Review by ujjwal-ab — APPROVED** [review]

**Pull request merged**